File: 每日一题/5.20/3355.零数组变换-i.py
#
# @lc app=leetcode.cn id=3355 lang=python3
#
# [3355] 零数组变换 I
#
from mytools import *
# @lc code=start
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        cover = [0] * (n + 1)
        for l, r in queries:
            cover[l] += 1
            if r + 1 < n:
                cover[r + 1] -= 1 
        for i in range(n):
            if i > 0:
                cover[i] += cover[i - 1]
        for i in range(n):
            if cover[i] < nums[i]: return False
        return True
# @lc code=end

File: 每日一题/2025-5/5.20/3355.零数组变换-i.cpp
/*
 * @lc app=leetcode.cn id=3355 lang=cpp
 *
 * [3355] 零数组变换 I
 */
#include "tools.h"
// @lc code=start
class Solution {
public:
    bool isZeroArray(vector<int>& nums, vector<vector<int>>& queries) {
        int n = nums.size();
        vector<int> cover(n, 0);
        for (int i = 0; i < queries.size(); i ++ )
        {
            cover[queries[i][0]] --;
            if (queries[i][1] + 1 < n) cover[queries[i][1] + 1] ++;
        }
        for (int i = 1; i < n; i ++ )
        {
            cover[i] += cover[i - 1];
        }
        for (int i = 0; i < n; i ++ )
        {
            if (nums[i] + cover[i] > 0) return 0;
        }
        return 1;
    }
};
// @lc code=end

